src/shim/IntersectionObserver.ts

```diff
@@ -1,5 +1,33 @@
 import global from './global';
 `!has('build-elide')`;
 import 'intersection-observer';
+import has from '../has/has';
 
-export default global.IntersectionObserver as typeof IntersectionObserver;
+const _intersectionObserver: typeof IntersectionObserver = global.IntersectionObserver as typeof IntersectionObserver;
+let replacement: typeof IntersectionObserver = _intersectionObserver;
+
+const Wrapper = class {
```

Comment:
This seems a bit verbose to have to do for each shim, also I don't think we need to wrap this outside of a test. I wonder if we could just essentially take your wrapping constructor idea and use it generically across the modules, kind of like a "live binding" wrapper to the global? That way we wouldn't even need a replace function per shim, they could just change the global?

```ts
function wrapper(nameOnGlobal: string) {
    return function(...args: any[]) {
        return new global[nameOnGlobal](...args);
    };
}
```
What `IntersectionObserver.ts` would then look like:
```ts
import global from './global';
`!has('build-elide')`;
import 'intersection-observer';
import has from '../has/has';
import wrapper from './somewhere/wrapper';

let value = global.IntersectionObserver;

if (has('test')) {
    value = wrapper('IntersectionObserver') ;
}

export default value as typeof IntersectionObserver;
```